ArduPlane/quadplane.cpp

```diff
@@ -2488,12 +2488,23 @@ int8_t QuadPlane::forward_throttle_pct(void)
     // inhibit reverse throttle and allow petrol engines with min > 0
     int8_t fwd_throttle_min = plane.have_reverse_thrust() ? 0 : plane.aparm.throttle_min;
     vel_forward.integrator = constrain_float(vel_forward.integrator, fwd_throttle_min, plane.aparm.throttle_max);
-    
-    // If we are below alt_cutoff then scale down the effect until it turns off at alt_cutoff and decay the integrator
-    float alt_cutoff = MAX(0,vel_forward_alt_cutoff);
-    float height_above_ground = plane.relative_ground_altitude(plane.g.rangefinder_landing);
-    vel_forward.last_pct = linear_interpolate(0, vel_forward.integrator,
-                                   height_above_ground, alt_cutoff, alt_cutoff+2);
+
+    if (in_vtol_auto() && is_vtol_land(plane.mission.get_current_nav_cmd().id) &&
+        (poscontrol.state == QPOS_POSITION1 || poscontrol.state == QPOS_POSITION2)) {
```

Comment:
It's getting aggressive but if you include a check of `motors->get_desired_spool_state() >= AP_Motors::DESIRED_GROUND_IDLE` that prevents ground throttle ups on the forward throttle. (It could actually be done after this entire if/else to add general protection in case the vehicle landed high somewhere). You can only get into this I think if you are starting the land after you are already on the ground, which you shouldn't do, but it's hard to predict what people do :)

Reply:
good idea, done